Assets/Scripts/Player/Stats.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Interfaces;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// This class is for testing the attack system implementation and does not have to be implemented like this.
+/// </summary>
+public class Stats : MonoBehaviour, IDamageable
+{
+    [Tooltip("The time the object should be marked as hit after being hit")]
+    [SerializeField]
+    private float hitMarkTime;
+    
+    // ------temp for hit animation------
+    private MeshRenderer _gameObjectRenderer;
+    private Material _prevMat;
+    private Material _hitMarkerMaterial;
+    // ----------
+
+    private float _timer;
+    private bool _hit;
+
+    private void Awake()
+    {
+        _gameObjectRenderer = gameObject.GetComponentInChildren<MeshRenderer>();
+        _hitMarkerMaterial = new Material(Shader.Find("Standard"));
+        _hitMarkerMaterial.color = Color.red;
+        // just put initial mat here
+        _prevMat = _gameObjectRenderer.material;
+    }
+
+    /// <summary>
+    /// Changes the objects color back to normal after being hit.
+    /// </summary>
+    private void Update()
+    {
+        if (_hit)
```

Comment:
```suggestion
        if (!_hit) return
```
This way it will skip the rest

Reply:
and the advantage is? skip one if statement? I rather take the readability, thanks 😄 (I changed it a bit)